Makefile

```diff
@@ -126,7 +126,17 @@ setup-bare: clean e2e.namespace
 E2E_NODES ?= 1
 E2E_FLAKE_ATTEMPTS ?= 1
 E2E_TIMEOUT ?= 90m
-E2E_OPTS ?= $(if $(E2E_SEED),-seed '$(E2E_SEED)') $(if $(TEST),-focus '$(TEST)') -flakeAttempts $(E2E_FLAKE_ATTEMPTS) -nodes $(E2E_NODES) -timeout $(E2E_TIMEOUT) -v -randomizeSuites -race -trace -progress
+E2E_COND_OPTS := $(if $(E2E_SEED),-seed '$(E2E_SEED)')
+# Optionally run an individual chunk of e2e test specs.
+# Do not use this from the CLI; this is intended to be used by CI only.
+E2E_TEST_CHUNK ?= all
+E2E_TEST_NUM_CHUNKS ?= 4
+ifeq (all,$(E2E_TEST_CHUNK))
+E2E_COND_OPTS := $(E2E_COND_OPTS) $(if $(TEST),-focus '$(TEST)')
+else
+E2E_COND_OPTS := $(E2E_COND_OPTS) -focus "$(shell go run ./test/e2e/split/... -chunks $(E2E_TEST_NUM_CHUNKS) -print-chunk $(E2E_TEST_CHUNK) ./test/e2e)"
```

Comment:
Just for my own edification: we're essentially firing off several e2e instances using the job matrix and printing out individual chunks (e.g. give me the chunk for the first index, second index, etc.), we create a kind cluster, ginkgo runs that chunk of focused tests, etc.? Any idea on whether this chunking is deterministic (it reads like it is when poking around the splitting package), and whether this scales out when adding new tests?

Reply:
Correct interpretation. Chunking is deterministic because specs are sorted, and scales with added or removed specs because test files are re-searched each run for new specs.